### PR TITLE
Integrate changes to Document.constant_equal

### PR DESCRIPTION
**Pull request by OlivierNicole**
This is part of the convergence work toward JSOO (see https://github.com/ocaml-wasm/wasm_of_ocaml/pull/57). It is based on ~~#67~~ and #70; the feature is contained in the last two commits.

It integrates https://github.com/ocsigen/js_of_ocaml/pull/1659 and adapts wasm_of_code to it.

### CODE REVIEW

File: compiler/lib/flow.ml [hunk 1]
```diff
@@ -321,6 +322,28 @@ let the_def_of info x =
         x
   | Pc c -> Some (Constant c)
 
+(* If [constant_identical a b = true], then the two values cannot be
+   distinguished, i.e., they are not different objects (and [caml_js_equals a b
+   = true]) and if both are floats, they are bitwise equal. *)
+let constant_identical a b =
```
Comment (vouillon) [inline compiler/lib/flow.ml:328]: We need to adapt this function since boxed values can be distinguished

Comment (OlivierNicole) [reply compiler/lib/flow.ml:328]: So I guess the boxed values in Wasm but not in JS are: Float, NativeString and String? In short everything but Int?

Comment (vouillon) [reply compiler/lib/flow.ml:328]: Right.

Comment (OlivierNicole) [reply compiler/lib/flow.ml]: Done

**Pull request merged**